duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt

```diff
@@ -184,15 +191,46 @@ class RealDuckPlayer @Inject constructor(
         pixelData: Map<String, String>,
     ) {
         if (!isFeatureEnabled) return
-        val duckPlayerPixelName = when (pixelName) {
-            "overlay" -> DUCK_PLAYER_OVERLAY_YOUTUBE_IMPRESSIONS
-            "play.use" -> DUCK_PLAYER_VIEW_FROM_YOUTUBE_MAIN_OVERLAY
-            "play.do_not_use" -> DUCK_PLAYER_OVERLAY_YOUTUBE_WATCH_HERE
-            else -> { null }
+        val duckPlayerPixelNames: List<Pair<DuckPlayerPixelName, PixelType>>? = when (pixelName) {
+            "overlay" -> {
+                listOf(
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_IMPRESSIONS to Count,
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_IMPRESSIONS_UNIQUE to Unique(),
+                )
+            }
+            "play.use" -> {
+                listOf(
+                    DUCK_PLAYER_VIEW_FROM_YOUTUBE_MAIN_OVERLAY to Count,
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_CHOICE_UNIQUE to Unique(),
+                )
+            }
+            "play.do_not_use" -> {
+                listOf(
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_WATCH_HERE to Count,
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_CHOICE_UNIQUE to Unique(),
+                )
+            }
+            "play.do_not_use.dismiss" -> {
+                listOf(
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_DISMISS to Count,
+                    DUCK_PLAYER_OVERLAY_YOUTUBE_CHOICE_UNIQUE to Unique(),
+                )
+            }
+            else -> {
+                Timber.d("Unknown duck player pixel: $pixelName, data: $pixelData")
+                null
+            }
         }
 
-        duckPlayerPixelName?.let {
-            pixel.fire(duckPlayerPixelName, parameters = pixelData)
+        duckPlayerPixelNames?.forEach { (duckPlayerPixelName, type) ->
+            val dataToSend = when {
+                duckPlayerPixelName == DUCK_PLAYER_OVERLAY_YOUTUBE_CHOICE_UNIQUE -> mapOf("choice" to pixelName.split(".").last())
```

Comment:
We shouldn't do this, at this point, I'd rather either create a wrapper class with `PixelName`, `Type`, and `params`, or just fire the necessary pixels from each `when` branch

Reply:
I agree, this is a little confusing. Encapsulating each in their own class seems like a better approach.